packages/core/src/routes/experience/classes/libraries/mfa-validator.ts

```diff
@@ -25,8 +28,30 @@ const mfaVerificationTypeToMfaFactorMap = Object.freeze({
   [VerificationType.WebAuthn]: MfaFactor.WebAuthn,
 }) satisfies Record<MfaVerificationType, MfaFactor>;
 
-const isMfaVerificationRecordType = (type: VerificationType): type is MfaVerificationType => {
-  return mfaVerificationTypes.includes(type);
+type MfaVerificationRecord = TotpVerification | WebAuthnVerification | BackupCodeVerification;
+
+const isMfaVerificationRecord = (
+  verification: VerificationRecord
+): verification is MfaVerificationRecord => {
+  return mfaVerificationTypes.includes(verification.type);
+};
+
+/**
+ * Check if the MFA verification record is a new bind MFA verification.
+ * New bind MFA verification can only be used for binding new MFA factors.
+ */
+const isNewBindMfaVerification = (verification: MfaVerificationRecord) => {
```

Comment:
nit: consider adding a variable to implicit mark the verification type (new or not)

Reply:
#6338

Reply:
#6339